### DISCUSSION TITLE
How to implement Modals in a reusable way?

### DISCUSSION THREAD

**Discussion by peersky** [chakra-ui/chakra-ui#8106] [Q&A]
I want my some of my  components for user input to be easily wrapped in to `Modal` with first class modal style right away. 

Hence when using `Modal` I want to have a common component (one modal per codebase), ideally that has "Submit" button in `ModalFooter`, which can trigger any submission logic that is defined within the `ModalBody`. 

Then I want to manage in a "context" approach such component to be able anywhere in the codebase to specify task like 

```jsx
const modalCtx = useContext(ModalContext) 
modalCtx.show("loginForm")
```
Or perhaps with wrapper:
```jsx
<ModalWrapper>
  <LoginForm/>
</ModalWrapper>
```

and within ModalContext/Wrapper l want to encapsulate the LoginForm from the rest of components:

```jsx
<ModalBody>
  currentModal === 'loginForm' && <LoginForm inModal={true} (???)>
</ModalBody>
<ModalFooter>
  <Button onClick={() => ????}>Submit</Button>
</ModalFooter>
```

More over I like to find solution where `LoginForm` 
- Does all login input data state management 
- Would not be dependant on Parent component nor the context (It can take a flag `inModal` to hide own buttons that would show up in non-modal implementation)
- Should be able to receive from Modal component trigger when submit button was pressed

But so far I could not figure out how to do such. I end up not using Footer/Titles that much, or I have to define components in a way that they are written only in modal implementation. 

Im wondering if anyone  has any tips on that?

**Comment by Joshua-Augment** [comment]
I ran into a similar issue, in the sense of reusable Modal components, as I like using Modals in my apps, so I created a little wrapper component for it, https://www.npmjs.com/package/chakraui-modal-wrapper, but its not built for state management, just to make handling Modals and its components easier, if that helps
